packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -29,71 +29,124 @@ import {
 			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
 			event.stopPropagation();
 		}
 	};
 
-	const closeURLPopover = () => {
-		setIsEditingLink( false );
-		setUrlInput( null );
-		setIsLinkOpen( false );
-	};
-
-	const autocompleteRef = useRef( null );
-
-	const onFocusOutside = ( event ) => {
-		const autocompleteElement = autocompleteRef.current;
-		if ( autocompleteElement && autocompleteElement.contains( event.target ) ) {
-			return;
+	/**
+	 * Updates the link attribute when it changes
+	 * through of the `onLinkChange` LinkControl callback.
+	 *
+	 * @param {Object|null} itemLink Link object if it has been selected. Otherwise, Null.
+	 */
+	const updateLink = ( itemLink ) => {
+		if ( ! itemLink ) {
+			// If not link, then set empty string to link attrs.
+			return setAttributes( { title: '', url: '' } );
 		}
-		closeURLPopover();
+
+		setAttributes( { title: itemLink.title, url: itemLink.url } );
 	};
 
-	const stopPropagation = ( event ) => {
-		event.stopPropagation();
+	/**
+	 * It updates the link attribute when the
+	 * link settings changes.
+	 *
+	 * @param {string} setting Setting type, for instance, `new-tab`.
+	 * @param {string} value Setting type value.
+	 */
+	const updateLinkSetting = ( setting, value ) => {
+		if ( 'new-tab' ) {
+			setAttributes( { opensInNewTab: value } );
+		}
 	};
 
-	const { label, url } = attributes;
 	let content;
 	if ( isSelected ) {
 		content = (
-			<TextControl
-				ref={ plainTextRef }
-				className="wp-block-navigation-menu-item__field"
-				value={ label }
-				onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
-				label={ __( 'Navigation Label' ) }
-				hideLabelFromVision={ true }
-			/>
+			<div className="wp-block-navigation-menu-item__field-container">
+				<TextControl
+					ref={ plainTextRef }
+					className="wp-block-navigation-menu-item__field"
+					value={ label }
+					onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
+					label={ __( 'Navigation Label' ) }
+					hideLabelFromVision={ true }
+				/>
+				{ isLinkOpen &&
+					<LinkControl
+						className="wp-block-navigation-menu-item__inline-link-input"
+						onKeyDown={ handleLinkControlOnKeyDown }
+						onKeyPress={ ( event ) => event.stopPropagation() }
+						currentLink={ link }
+						onLinkChange={ updateLink }
+						onClose={ () => {
+							setWasCloseByLinkControl( true );
+							setIsLinkOpen( false );
+						} }
+						currentSettings={ { 'new-tab': opensInNewTab } }
+						onSettingsChange={ updateLinkSetting }
+						fetchSearchSuggestions={ fetchSearchSuggestions }
+					/>
+				}
+			</div>
 		);
 	} else {
-		content = <div className="wp-block-navigation-menu-item__container">
-			{ label }
-		</div>;
+		content = (
+			<div className="wp-block-navigation-menu-item__container">
+				{ ( link && link.url ) && (
+					<ExternalLink href={ link.url }>{ label }</ExternalLink>
```

Comment:
We should probably only use that when it's actually an external link. Currently it marks internal links, too.

Reply:
I don't think there should be any links in the `edit` interface of the block. 

I think it's a similar case to the button block, where in the editor the button is not actually clickable as a link.

The text content of the block has since been refactored to a `RichText` in `master`, and I think it should just stay as that whether the block is selected or deselected.